### PR TITLE
Feat/homepage mobile UI

### PR DESCRIPTION
**Pull request by hansac91**

### CODE REVIEW

**Review by NerdbordMentor — APPROVED** [review]
## Code Review Report

### Task goal
The task goal was to create a "My profile" UI using already created components. The UI should be implemented as close as possible to the provided design.

### Code Changes Review

#### app/globals.scss
The removal of `overflow-x: hidden;` from the body styling is fine. This change doesn't seem to affect the task requirements directly.

#### src/components/Dropdowns/DropdownFilter/DropdownFilter.module.scss
The changes in this file are mostly related to mobile responsiveness. The dropdown menu now has a fixed position on mobile screens and takes up the full width of the screen. This seems to be in line with the task requirements.

#### src/components/Dropdowns/DropdownFilterMulti/DropdownFilterMulti.tsx
The changes in this file also seem to be related to mobile responsiveness. The dropdown menu now has different classes depending on whether it's active and whether the screen is a mobile screen. This is in line with the task requirements.

#### src/components/Filters/Filters.module.scss
The changes in this file are also related to mobile responsiveness. The filters now have a horizontal scroll on mobile screens. This seems to be in line with the task requirements.

#### src/components/Headers/DefaultHeader/DefaultHeader.module.scss and src/components/Headers/DefaultHeader/DefaultHeader.tsx
The changes in these files are related to the "Create profile" and "My profile" buttons. On mobile screens, these buttons now have a fixed position at the bottom of the screen. This is in line with the task requirements.

#### src/components/ProfileList/ProfileList.module.scss and src/components/ProfileList/ProfileListItem.tsx
The changes in these files are related to the profile list items. On mobile screens, there is now only one profile list item per row, and the width of each item is fixed. The number of technologies displayed for each item also depends on the screen width. These changes are in line with the task requirements.

### Conclusion
The code changes seem to cover the task requirements. The "My profile" UI has been updated to match the provided design as closely as possible, and the changes are mostly related to making the UI responsive on mobile screens.

File: src/components/Dropdowns/DropdownFilterMulti/DropdownFilterMulti.tsx [hunk 1]
```diff
@@ -56,7 +67,18 @@ export const DropdownFilterMulti = ({
           {arrow === 'IoIosArrowUp' ? <IoIosArrowUp /> : <IoIosArrowDown />}
         </button>
         {isDropdownActive && (
-          <div className={styles.dropdown}>
+          <div className={`${styles.dropdown} ${dropdownClass}`}>
+              <div className={styles.titleContainer}>
+              <div className={styles.dropdownTitle}>
+              {text}
```
Comment (Sumick) [inline src/components/Dropdowns/DropdownFilterMulti/DropdownFilterMulti.tsx:73]: **nitpick:** Hmm, looks like it's missing formatting

**Review by Sumick — APPROVED** [review]
There is some issue with formatting in few places, can you check it before merging? Overall good job!

**Review by NerdbordMentor — APPROVED** [review]
## Code Review Report

### Task goal
Adjust Homepage UI, so it's able to display on mobile devices.

### Requirements
UI should be implemented as close as possible to provided design.

### Review

#### app/globals.scss
The removal of `overflow-x: hidden;` is acceptable as it may have been preventing horizontal scrolling on mobile devices.

#### src/components/Button/Button.module.scss
The addition of the `.buttonTertiary` class seems to be for a new button variant that is only visible on mobile devices. This is in line with the task requirements.

#### src/components/Button/Button.tsx
The addition of the 'tertiary' button variant is consistent with the changes in the SCSS file. The use of `useMemo` for determining the button class name is a good practice for performance optimization.

#### src/components/Dropdowns/DropdownFilter/DropdownFilter.module.scss
The changes made to the dropdown component seem to be for better mobile responsiveness. The dropdown now takes up the full width of the screen on mobile devices and appears from the bottom. This is a common design pattern for mobile interfaces.

#### src/components/Dropdowns/DropdownFilter/DropdownFilter.tsx
The addition of an "Apply" button in the dropdown is a good UX improvement for mobile users. It makes it clear how to close the dropdown after making a selection.

#### src/components/Dropdowns/DropdownFilterMulti/DropdownFilterMulti.tsx
Similar changes to `DropdownFilter.tsx` were made here for the multi-select dropdown. This ensures consistency across different types of dropdowns.

#### src/components/Filters/Filters.module.scss
The changes here make the filters horizontally scrollable on mobile devices. This is a common design pattern for mobile interfaces and is in line with the task requirements.

#### src/components/Headers/DefaultHeader/DefaultHeader.module.scss
The addition of the `.mobileButton` class seems to be for a new button that is only visible on mobile devices. This is in line with the task requirements.

#### src/components/Headers/DefaultHeader/DefaultHeader.tsx
The removal of the profile button from the header is acceptable if it was moved elsewhere in the mobile design.

#### src/components/ProfileList/ProfileList.module.scss
The changes here make the profile list more responsive on mobile devices. The profiles now stack vertically on mobile, which is a common design pattern for mobile interfaces.

#### src/components/ProfileList/ProfileListItem.tsx
The changes here make the tech stack list in each profile item more responsive. On mobile devices, fewer technologies are displayed initially, which is a good UX improvement for smaller screens.

### Conclusion
The code changes are in line with the task requirements. The UI has been adjusted to be more responsive on mobile devices, and the changes seem to closely follow the provided design.

**Review by wawrzonnn — CHANGES_REQUESTED** [review]
When the filter window is active, the background should be blurred

**Comment by wawrzonnn** [comment]
> When the filter window is active, the background should be blurred

hania

**Review by wawrzonnn — APPROVED** [review]
Approved. However, a few suggestions:
1. We don't need the `isOverlayActive` state since it will always mirror the value of `isDropdownActive`. We can just use one of them for simplicity.
2. Using a switch statement here seems like overkill. An `if-else` construct would suffice and make the code more concise.

**Pull request merged**